### ISSUE TITLE
packaging: build failures for raspbian/buster

### ISSUE THREAD

**Issue by patrick-stephens** [fluent/fluent-bit#6175]
## Bug Report

Raspbian buster containers are triggering a strange sub-reaper fatal error immediately on attempted build: https://github.com/fluent/fluent-bit/actions/runs/3221397977/jobs/5269286724#step:6:1636

```
+ docker run -v /home/runner/work/fluent-bit/fluent-bit/packaging/packages/raspbian/buster/master/staging/:/output flb-raspbian/buster
[FATAL tini (27)] PR_SET_CHILD_SUBREAPER is unavailable on this platform. Are you using Linux >= 3.4?
+ echo 'Could not compile using image flb-raspbian/buster'
Could not compile using image flb-raspbian/buster
```

I am unable to reproduce locally so suspect some specific set up on the Github runner.

Now, delving into the set up shows that we are using the deprecated `resin/rpi-raspbian:buster ` image for raspbian/buster but not for raspbian/bullseye which works and uses `balenalib/rpi-raspbian:buster`

One of the main differences between the two is that `tini` is no longer used which would therefore remove this weird subreaper error.

Unfortunately using the `balenalib/rpi-raspbian:buster` image triggers a separate `cmake` issue during build as it uses a more recent version of cmake. Pinning cmake to the working 3.13.4-1 version used in the old `resin/rpi-raspbian:buster ` image seems to work so seems to be the solution.

TLDR; Update to `balenalib/rpi-raspbian:buster` image to remove `tini` but then pin `cmake` version to old 3.13.4-1 version.

```diff
diff --git a/packaging/distros/raspbian/Dockerfile b/packaging/distros/raspbian/Dockerfile
index 1c44e3425..5268747d4 100644
--- a/packaging/distros/raspbian/Dockerfile
+++ b/packaging/distros/raspbian/Dockerfile
@@ -6,17 +6,16 @@ ARG BASE_BUILDER
 # Lookup the name to use below but should follow the '<distro>-base' convention with slashes replaced.
 # Use buildkit to skip unused base images: DOCKER_BUILDKIT=1
 
-# raspbian/buster base image
-# Use deprecated image as it pins dependencies to working ones.
-# Current balenalib/rpi-raspbian:buster triggers cmake failures for GNU C compiler detection
-FROM resin/rpi-raspbian:buster as raspbian-buster-base
+FROM balenalib/rpi-raspbian:buster as raspbian-buster-base
 ENV DEBIAN_FRONTEND noninteractive
 
 # Builder image so dependencies can be latest, recommended and no need to wipe
+# We pin cmake to a working version (latest 3.16 triggers cmake failures for GNU C compiler detection)
 # hadolint ignore=DL3008,DL3015
 RUN apt-get update && \
-    apt-get install -y curl ca-certificates build-essential \
-    cmake make bash sudo wget unzip dh-make \
+    apt-get install -y cmake=3.13.4-1 cmake-data=3.13.4-1 \
+    curl ca-certificates build-essential \
+    make bash sudo wget unzip dh-make \
     libsystemd-dev zlib1g-dev flex bison \
     libssl1.1 libssl-dev libpq-dev postgresql-server-dev-all \
     libsasl2-2 libsasl2-dev libyaml-dev && \
```

**Any PR must test this raspbian/buster build in CI using the appropriate workflow to confirm the issue is resolved.**

**Comment by patrick-stephens** [comment]
This needs merging to 1.9 as well.
